Lib/test/libregrtest/main.py

```diff
@@ -534,7 +536,13 @@ def _main(self, tests, kwargs):
             self.rerun_failed_tests()
 
         self.finalize()
-        sys.exit(len(self.bad) > 0 or self.interrupted)
+        if self.bad:
+            sys.exit(2)
+        if self.interrupted:
+            sys.exit(3)
+        if self.ns.fail_env_changed and self.environment_changed:
```

Comment:
Is the if stmt of line 543:
line 543: `if self.ns.fail_env_changed and self.environment_chnged:`
line 477: `if self.environment_changed and self.ns.fail_env_changed:`

not about the behavior, but would it be better to be the same sequence about readability?